File: test-suite/src/test/java/io/micronaut/kafka/docs/consumer/sendto/SendToProductListenerTest.java
package io.micronaut.kafka.docs.consumer.sendto;

import io.micronaut.context.ApplicationContext;
import io.micronaut.kafka.docs.Product;
import org.junit.jupiter.api.Test;

import java.util.Map;

import static java.util.concurrent.TimeUnit.SECONDS;
import static org.awaitility.Awaitility.await;

class SendToProductListenerTest {

    @Test
    void testSendProduct() {
        try (ApplicationContext ctx = ApplicationContext.run(
            Map.of("kafka.enabled", "true", "spec.name", "SendToProductListenerTest")
        )) {
            Product product = new Product("Blue Trainers", 5);
            ProductClient client = ctx.getBean(ProductClient.class);
            client.send("Nike", product);
            QuantityListener listener = ctx.getBean(QuantityListener.class);
            await().atMost(10, SECONDS).until(() ->
                listener.quantity != null &&
                listener.quantity == 5
            );
        }
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/kafka/docs/consumer/sendto/SendToProductListenerTest.kt
package io.micronaut.kafka.docs.consumer.sendto

import io.micronaut.context.ApplicationContext
import io.micronaut.kafka.docs.Product
import org.awaitility.Awaitility
import org.junit.jupiter.api.Test
import java.util.Map
import java.util.concurrent.TimeUnit

class SendToProductListenerTest {

    @Test
    fun testSendProduct() {
        ApplicationContext.run(
            Map.of<String, Any>("kafka.enabled", "true", "spec.name", "SendToProductListenerTest")
        ).use { ctx ->
            val product = Product("Blue Trainers", 5)
            val client = ctx.getBean(ProductClient::class.java)
            client.send("Nike", product)
            val listener = ctx.getBean(QuantityListener::class.java)
            Awaitility.await().atMost(10, TimeUnit.SECONDS).until {
                listener.quantity != null &&
                listener.quantity!!.toInt() == 5
            }
        }
    }
}
